src/main/editDelete/preProcess.ts

```diff
@@ -50,6 +31,74 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
   fileName: 'PLACEHOLDER FILENAME',
 });
 
+const constructWord = (
+  word: string,
+  startTime: number,
+  duration: number,
+  outputStartTime: number,
+  key: string
+) => {
+  const newWord: Word = {
+    word,
+    startTime: roundToMs(startTime),
+    duration: roundToMs(duration),
+    outputStartTime: roundToMs(outputStartTime),
+    deleted: false,
+    key,
+    fileName: 'PLACEHOLDER FILENAME',
+  };
+
+  return newWord;
+};
+
+/**
+ * Adds spaces between words which represent the silence between each word
+ * @param result The output being constructed
+ * @param word The current element of words
+ * @param index The index of word in words
+ * @param words The list of words being reduced
+ * @returns The updated transcript with a silence after word
+ */
+let TOTAL_DURATION = 0;
+const addSpaces = (
```

Comment:
Maybe make a helper type like we have for MapCallback and give it that: i.e.

`type ReduceCallback<T, U> = (acc: T[], curr: T, index: number; arr: T[]) => U;`

Then the type of this function can just be `ReduceCallback<Word, Word[]>` i.e. it takes a list containing Word objects and returns another list of word objects

Nice work using reduce by the way! It's a tricky one to understand

Reply:
Ahhh I see how that words now. Wouldn't have figured that one out myself (I tried lol). Addressed here: https://github.com/patrickbrett/mlvet/pull/54/commits/8edf6d64c9d50d8fba9ae4f007537ad444542a37